src/simplewallet/simplewallet.cpp

```diff
@@ -1090,6 +1074,71 @@ bool simple_wallet::init(const boost::program_options::variables_map& vm)
       bool r = new_wallet(vm, m_recovery_key, m_restore_deterministic_wallet, m_non_deterministic, old_language);
       CHECK_AND_ASSERT_MES(r, false, tr("account creation failed"));
     }
+    if (!m_restore_height && m_restoring)
+    {
+      uint32_t version;
+      if (!try_connect_to_daemon(false, &version))
+      {
+        fail_msg_writer() << tr("failed to connect to the daemon");
+        return false;
+      }
+      while (true)
+      {
+        std::string heightstr = command_line::input_line(
+          version < 0x1006 ? "Restore from specific blockchain height (optional, default 0): ":
+                             "Restore from specific blockchain height (optional, default 0),\nor alternatively from specific date (YYYY-MM-DD): ");
+        if (std::cin.eof())
+          return false;
+        if (heightstr.empty())
+        {
+          m_restore_height = 0;
+          break;
+        }
+        try
+        {
+          m_restore_height = boost::lexical_cast<uint64_t>(heightstr);
+          break;
+        }
+        catch (boost::bad_lexical_cast &)
+        {
+          if (version < 0x1006)
+          {
+            fail_msg_writer() << tr("bad m_restore_height parameter: ") << heightstr;
```

Comment:
I believe the error message is correct, since the user is either offline or connected to an older RPC (where the date format is unsupported) and is passing whatever string that doesn't parse as the block height. If we're not connected to a recent enough server and pass a string "2017-01-13", it should be considered as a bad parameter.